### PR TITLE
Support skip logging for specific path

### PR DESCRIPTION
**Pull request by ota42y**
Gin's logger support ignore logging in specific path.
gin-gonic/gin#1798 (comment)

So we want to add this feature.

### CODE REVIEW

File: zap.go [hunk 1]
```diff
@@ -24,16 +31,30 @@ import (
...
+	}
+
 	return func(c *gin.Context) {
 		start := time.Now()
 		// some evil middlewares modify this values
 		path := c.Request.URL.Path
 		query := c.Request.URL.RawQuery
 		c.Next()
 
+		if skipPaths[path] {
```
Comment (appleboy) [inline zap.go]: can't return if match the skip paths.

See: https://github.com/gin-contrib/logger/blob/55c0f3c3242cfb91c355b151caaf8092528ff03b/logger.go#L42-L102

**Review by appleboy — CHANGES_REQUESTED** [review]

File: zap.go (hunk 1 above)
Comment (ota42y) [reply zap.go]: Sorry, I can't understand why we can't return.

I think gin-contrib/logger support `SkipPathRegexp` so we can't return.
But this changes doesn't support regexp so we can return.
Does this change require regexp feature? (I think it is not difficult :) )

Comment (mmwtsn) [reply zap.go]: @ota42y could you mirror the implementation in the main Gin logger here?

https://github.com/gin-gonic/gin/blob/34ce2104cad324f444943c528746bf6d23643cd3/logger.go#L243

This function should be pretty similar to that when it comes to the skip behavior.

Comment (ota42y) [reply zap.go]: Fixed 🙋

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
@ota42y Tanks for your contribution.

**Pull request merged**